### PR TITLE
clientConfig loader now load defaults

### PR DESCRIPTION
**Pull request by mrkwtz**
This change allows the standard defaults when it comes to client configuration to be honored, like the KUBECONFIG env

### CODE REVIEW

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: kubernetes/utils.go [hunk 1]
```diff
@@ -176,12 +176,11 @@ func getK8Config(ctx context.Context, d *plugin.QueryData) (clientcmd.ClientConf
 	kubernetesConfig := GetConfig(d.Connection)
 
 	// Set default loader and overriding rules
-	loader := &clientcmd.ClientConfigLoadingRules{}
+	loader := clientcmd.NewDefaultClientConfigLoadingRules()
```
Comment (cbruno10) [inline kubernetes/utils.go:179]: @mrkwtz @bigdatasourav - I'm not familiar with the default loader, but would either of you be able to please explain what environment variables this plugin now looks at and in which order? I think we should also update all of the references to file locations in `config/kuberenetes.spc`, `docs/index.md`, and the `README.md` (if there any here), so users can better understand what this plugin supports.

**Review by cbruno10 — COMMENTED** [review]
@mrkwtz Please see question/suggestions, if you have any questions, please let me know, thanks!

File: kubernetes/utils.go (hunk 1 above)
Comment (bigdatasourav) [reply kubernetes/utils.go:179]: Hey @mrkwtz, Could you please update the reference files mentioned above with how the KUBECONFIG env variable will work and in which order?

**Comment by cbruno10** [comment]
Hi @mrkwtz , if you're still interested in working together on this PR, can you please see my question in https://github.com/turbot/steampipe-plugin-kubernetes/pull/92#discussion_r1016781767? We're interested in this PR and would like to learn more about client configuration.

Thanks!

**Comment by mrkwtz** [comment]
Hi @cbruno10 thanks for keeping that open and asking :) I would still love to complete this but am missing time right now to do that. It's still an item on my to-do list :)

**Comment by cbruno10** [comment]
@mrkwtz If you're ok with us helping get this PR over the line, please let us know and we can either raise a PR against your forked branch, or merge your branch into a new feature branch in this repo and then make any changes on top of it.

**Comment by bigdatasourav** [comment]
@mrkwtz, I have raised a [PR](https://github.com/turbot/steampipe-plugin-kubernetes/pull/142) that includes updated documentation regarding the support of the KUBECONFIG environment variable. Merging this PR into that one will consolidate the changes and make tracking easier. Please provide your valuable suggestions and feedback regarding the modifications I've made. Your input would be greatly appreciated.

**Comment by mrkwtz** [comment]
@bigdatasourav in https://github.com/turbot/steampipe-plugin-kubernetes/pull/92#discussion_r1017969869 you advised to add a order of usage. That's why I didn't have time to complete the PR. I again took a look at the code and I would assume (not tested) that the order is like this 

1. kubernetesConfig.ConfigPath
2. kubernetesConfig.ConfigPaths
3. KUBE_CONFIG_PATHS env
4. KUBERNETES_MASTER env
5. KUBECONFIG env

So the KUBECONFIG env is a fallback in case none of the above have been configured. That's the conclusion because it's mostly the flow of the code (all the elseif statements) and if something is configured (1-4) it's added as an override when the kubeclient is created. You may want to add that information but beforehand test if that's really the case :)

**Pull request merged**